core/src/main/scala/org/apache/spark/internal/io/FileCommitProtocol.scala

```diff
@@ -232,6 +253,89 @@ object FileCommitProtocol extends Logging {
   def getStagingDir(path: String, jobId: String): Path = {
     new Path(path, ".spark-staging-" + jobId)
   }
+
+  def externalTempPath(
+      path: Path,
+      hadoopConf: Configuration,
+      stagingDir: String,
+      engineType: String,
+      jobId: String): Path = {
+    val extURI = path.toUri
+    if (extURI.getScheme == "viewfs") {
```

Comment:
always a bit brittle using the uri scheme. any way to avoid?

Reply:
> always a bit brittle using the uri scheme. any way to avoid?

To be honest, I don't know why we need to handle viewfs separately and after searching origin pull request of this part, don't see the discussion, so I am not sure about this.  And this part is just moved to here.

Reply:
probably there to ensure you don't create staging dirs in a different fs

Reply:
> probably there to ensure you don't create staging dirs in a different fs

Yea, I remembered that hive will check tempLocation and targetOutputPath's FS and EZ when call HiveMetastore loadTable/loadPartition, Should move that logical to here

Reply:
> probably there to ensure you don't create staging dirs in a different fs

How about current?